File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/events/CommunicationUIEventCode.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.android.communication.ui.configuration.events;


import com.azure.android.core.util.ExpandableStringEnum;

import java.util.Collection;

/**
 * Defines values for CommunicationUIEventCode.
 */
public class CommunicationUIEventCode extends ExpandableStringEnum<CommunicationUIEventCode> {
    public static final CommunicationUIEventCode NOP = fromString("nop");
    public static final CommunicationUIEventCode CALL_EVICTED = fromString("callEvicted");

    /**
     * Creates or finds a CommunicationUIEventCode from its string representation.
     *
     * @param name a name to look for.
     * @return the corresponding CommunicationUIEventCode.
     */
    private static CommunicationUIEventCode fromString(final String name) {
        return fromString(name, CommunicationUIEventCode.class);
    }

    /**
     * @return known CommunicationUIEventCode values.
     */
    public static Collection<CommunicationUIEventCode> values() {
        return values(CommunicationUIEventCode.class);
    }
}

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/models/CommunicationUIEventCode.kt
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package com.azure.android.communication.ui.calling.models

import com.azure.android.core.util.ExpandableStringEnum

/**
 * Defines values for CommunicationUIEventCode.
 */
internal class CommunicationUIEventCode : ExpandableStringEnum<CommunicationUIEventCode?>() {
    companion object {
        val CALL_EVICTED = fromString("callEvicted", CommunicationUIEventCode::class.java)
    }

    /**
     * Creates or finds a CommunicationUIEventCode from its string representation.
     *
     * @param name a name to look for.
     * @return the corresponding CommunicationUIEventCode.
     */
    private fun fromString(name: String): CommunicationUIEventCode {
        return fromString(name, CommunicationUIEventCode::class.java)
    }

    /**
     * @return known CommunicationUIEventCode values.
     */
    fun values(): Collection<CommunicationUIEventCode> {
        return values(CommunicationUIEventCode::class.java)
    }
}
